src/ArmTemplates/Extractor/Utilities/DataProcessors/ProductApiDataProcessor.cs

```diff
@@ -0,0 +1,66 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.ProductApis;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors.Absctraction;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors
+{
+    public class ProductApiDataProcessor: IProductApiDataProcessor
+    {
+        public IDictionary<string, string> OverrideRules { get; }
+        public ProductApiDataProcessor() {
+            this.OverrideRules = new Dictionary<string, string>() {
+                { "Starter", "starter" },
+                { "Unlimited", "unlimited" }
+            };
+        }
+
+        public void ProcessData(List<ProductApiTemplateResource> productApiTemplates, ExtractorParameters extractorParameters)
+        {
+
+            if (productApiTemplates.IsNullOrEmpty())
+            {
+                return;
+            }
+
+            foreach (var productApiTemplate in productApiTemplates)
+            {
+                // save Original name for future references
+                productApiTemplate.OriginalName = productApiTemplate.Name;
+                productApiTemplate.NewName = productApiTemplate.Name;
+                
+                if (extractorParameters.OverrideProductGuids)
+                {
+                    this.OverrideName(productApiTemplate);
+                }
+            }
+
+        }
+
+        public void OverrideName(ProductApiTemplateResource template)
+        {
+            if (this.OverrideRules == null)
```

Comment:
1. I think this is not needed since we are defining `OverrideRules` in constructor explicitly
2. I am trying to use `IsNullOrEmpty()` [extension method](https://github.com/Azure/azure-api-management-devops-resource-kit/blob/43e6806f162bd96a1f7debe23861aacc3e7f17e4/src/ArmTemplates/Common/Extensions/CollectionExtensions.cs#L13) everywhere, because if `OverrideRules` collection is not null, but is empty, there is still no reason to move further. I am not sure if it is okay, that I am using that extension elsewhere, so you can suggest smth else! 

This comment applies to other DataProcessors also

Reply:
Useful extension :) Replaced comparing with IsNullOrEmpty.